lib/child-writer/index.js

```diff
@@ -170,14 +182,112 @@ exports.write = (image, drive, options) => {
     }
   }
 
+  // Timeout function that only runs if we exceed the specified timeout,
+  // terminating the server.
+  // @type {Timeout}
+  let terminationTimeout = null
+
+  // If the child process is created successfully it will be stored here.
+  // @type {ChildProcess}
+  let maybeChild = null
+
+  // Whether the child process is running
+  // @type {Boolean}
+  let isChildRunning = null
```

Comment:
Is there a way we can reduce the amount of mutable variables tracking state that we have here? It obviously works, but I feel its a matter of time until we will get weird edge cases where these variables are not updated appropriately, and that will be really hard to debug.